docs/dotnet/Runner.md

```diff
@@ -0,0 +1,136 @@
+ - `time_limit` - soft (see [Methodology.md](../Methodology.md)) time limit for benchmarking the computation of either the objective function or its gradient/Jacobian.
+ - `-rep` *(applicable only for GMM)* - if enabled, all input data points are expected to have one shared value.
+
+## Adding new modules
+(see [Modules.md](./Modules.md))
+
+## Adding new objective types
+
+ 1. Define `TInput`, `TOutput`, and, if necessary, `TParameters` types in the `/src/dotnet/runner/Data/TData.cs` file where `T` is the name of the new objective. Keep them in the `DotnetRunner.Data` namespace.
+
+    `TInput` is the data type for the new objective's inputs, `TOutput` is a structure that contains outputs of both new objective function and its target Jacobian.
+    `TParameters` is a data type containing all configurable parameters of the benchmark. If there are no such parameters then it can be omitted.
+
+ 2. In the `ModuleLoader` class (`/src/dotnet/runner/ModuleLoader.cs`) add the following function:
+    ```csharp
+    public ITest<GMMInput, GMMOutput> GetTTest()
+    {
+        if (container.TryGetExport(out ITest<TInput, TOutput> tTest))
+            return tTest;
+        else
+            throw new InvalidOperationException("The specified module doesn't support the T objective.");
+    }
+    ```
+    Where `T` still is the name of the new objective.
+
+ 3. Create `/src/dotnet/runner/Benchmarks/TBenchmark.cs` with the following content:
+    ```csharp
+    using DotnetRunner.Data;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+
+    namespace DotnetRunner.Benchmarks
+    {
+        public class TBenchmark : Benchmark<TInput, TOutput, TParameters>
+        {
+            protected override TInput ReadInputData(string inputFilePath, TParameters parameters)
+            {
+                // implementation
+            }
+
+            protected override ITest<TInput, TOutput> GetTest(ModuleLoader moduleLoader)
+            {
+                return moduleLoader.GetBATest();
+            }
+
+            protected override void SaveOutputToFile(TOutput output, string outputPrefix, string input_basename, string module_basename)
+            {
+                // implementation
+            }
+        }
+    }
+    ```
+
+    If you opted to omit defining `TParameters` type on step one, then inherit from `Benchmark<TInput, TOutput>` instead. `ReadInputData` in that case would have the following signature:
+    ```csharp
+    protected override TInput ReadInputData(string inputFilePath, DefaultParameters parameters)
+    ```
+
+    Implement the methods so that they would:
+
+    - 
+      ```csharp
+      protected override TInput ReadInputData(string inputFilePath, TParameters parameters)
+      ```
+          
+      Open input_file and loads data to the structure of the TInput type. 
```

Comment:
Seems like a misprint `input_file` instead of `input file`

Reply:
Fixed.